src/term/transform/linearize.rs

```diff
@@ -181,13 +169,23 @@ fn term_to_affine(term: &mut Term, var_uses: &mut HashMap<Name, Val>, let_bodies
         uses => {
           term_to_affine(val, var_uses, let_bodies);
           term_to_affine(nxt, var_uses, let_bodies);
-          duplicate_let(nam, nxt, uses, val);
+
+          let mut new_uses = actual_var_use(var_uses, nam);
+
+          // If the number of uses changed (because a term was linearized out):
+          //  We could redo the whole pass on the term.
+          //  Or we can just create extra half-erased-dups `let {nam_n *} = nam_m_dup; nxt` to match the correct labels
```

Comment:
We don't want to create unnecessary dups since they decrease performance.

Reply:
Won't the pre_reduce pass remove them anyway? Or should find a better way to not generate on the first place?

Reply:
pre_reduce will not remove them since they are not in active position. If we know that an inet package is normalized, it's safe to remove any half-erased dups, but that would have to be done in a separate pass.

Overall, it'd be better to not generate them in the first place, then we don't have to worry about removing them. I think this is the only place that generates them, so we haven't needed this additional pass so far.

Reply:
`pre_reduce will not remove them since they are not in active position.`
wdym?
![image](https://github.com/HigherOrderCO/hvm-lang/assets/19491161/464882fb-0ed8-4e4d-950a-4e11db2926cc)